src/types.rs

```diff
@@ -619,6 +621,50 @@ impl From<AcmeRenewalInfoCertIdentifier<'_>> for EncodedCertificateIdentifier {
     }
 }
 
+#[cfg(feature = "x509-parser")]
+impl<'a> TryFrom<&'a [u8]> for EncodedCertificateIdentifier {
+    type Error = String;
+
+    fn try_from(certificate_der: &'a [u8]) -> Result<Self, Self::Error> {
+        let (_, parsed_cert) = x509_parser::parse_x509_certificate(certificate_der)
```

Comment:
Nit: let's import `parse_x509_certificate()` and `ParsedExtension` directly?

Reply:
Done.